src/govuk/components/modal-dialogue/modal-dialogue.js

```diff
@@ -0,0 +1,186 @@
+import { nodeListForEach } from '../../common'
+import '../../vendor/polyfills/Element/prototype/classList'
+import '../../vendor/polyfills/Function/prototype/bind'
+import '../../vendor/polyfills/Event' // addEventListener and event.target normaliziation
+
+function ModalDialogue ($module) {
+  this.$module = $module
+  this.$dialogBox = $module.querySelector('dialog')
+  this.$container = document.documentElement
+
+  // Check for browser support
+  this.hasNativeDialog = 'showModal' in this.$dialogBox
+
+  // Allowed focussable elements
+  this.focussable = [
+    'button',
+    '[href]',
+    'input',
+    'select',
+    'textarea'
+  ]
+}
+
+// Initialize component
+ModalDialogue.prototype.init = function (options) {
+  this.options = options || {}
+
+  this.open = this.handleOpen.bind(this)
+  this.close = this.handleClose.bind(this)
+  this.focus = this.handleFocus.bind(this)
+  this.focusTrap = this.handleFocusTrap.bind(this)
+  this.boundKeyDown = this.handleKeyDown.bind(this)
+
+  // Modal elements
+  this.$buttonClose = this.$dialogBox.querySelector('.govuk-modal-dialogue__close')
+  this.$focussable = this.$dialogBox.querySelectorAll(this.focussable.toString())
+  this.$focusableLast = this.$focussable[this.$focussable.length - 1]
+  this.$focusElement = this.options.focusElement || this.$dialogBox
+
+  if (this.$dialogBox.hasAttribute('open')) {
+    this.open()
+  }
+
+  this.initEvents()
+
+  return this
+}
+
+// Initialize component events
+ModalDialogue.prototype.initEvents = function (options) {
+  if (this.options.triggerElement) {
+    this.options.triggerElement.addEventListener('click', this.open)
+  }
+
+  // Close dialogue on close button click
+  this.$buttonClose.addEventListener('click', this.close)
+}
+
+// Open modal
+ModalDialogue.prototype.handleOpen = function (event) {
+  if (event) {
+    event.preventDefault()
+  }
+
+  // Save last-focussed element
+  this.$lastActiveElement = document.activeElement
+
+  // Disable scrolling, show wrapper
+  this.$container.classList.add('govuk-!-scroll-disabled')
```

Comment:
Does `govuk-!-scroll-disabled` do anything? I can't see any styles created for it.